HW4_Bredov/protein_analyzer_tool.py

```diff
@@ -0,0 +1,353 @@
+from typing import Iterable, Tuple
+
+AA_TR_DICT = {'Ala': 'A', 'Arg': 'R', 'Asn': 'N', 'Asp': 'D',
+              'Cys': 'C', 'Gln': 'E', 'Glu': 'Q', 'Gly': 'G',
+              'His': 'H', 'Ile': 'I', 'Leu': 'L', 'Lys': 'K',
+              'Met': 'M', 'Phe': 'F', 'Pro': 'P', 'Ser': 'S',
+              'Thr': 'T', 'Trp': 'W', 'Tyr': 'Y', 'Val': 'V'}
+AA_UNIPROT_CONTENT = {
+"A": 9.03, "R": 5.84, "N": 3.79, "D": 5.47, "C": 1.29,
+"Q": 3.80, "E": 6.24, "G": 7.27, "H": 2.22, "I": 5.53,
+"L": 9.85, "K": 4.93, "M": 2.33, "F": 3.88, "P": 4.99,
+"S": 6.82, "T": 5.55, "W": 1.30, "Y": 2.88, "V": 6.86
+}
+AA_CHARGES = {
+"A": 0, "R": 1, "N": 0, "D": -1, "C": 0,
+"Q": 0, "E": -1, "G": 0, "H": 1, "I": 0,
+"L": 0, "K": 1, "M": 0, "F": 0, "P": 0,
+"S": 0, "T": 0, "W": 0, "Y": 0, "V": 0
+}
+
+
+def Mann_Whitney_U(seq1: Iterable[int], seq2: Iterable[int]) -> bool:
+    """
+    Mann-Whitney U-test. Used to compare aminoacids composition in given sequence with average protein composition provided by Uniprot.
+    Used as a second step of checkup in `check_and_procees_seq` function if sequence uses 1-letter abbreviation.
+    """
+    len_seq1, len_seq2 = len(seq1), len(seq2)
+    r1, r2 = dict.fromkeys(map(str, seq1), 0), dict.fromkeys(map(str, seq2), 0)
+
+    r = sorted(list(seq1) + list(seq2))
+    r_dict = dict.fromkeys(map(str, r), ())
+    
+    for index, value in enumerate(r):
+        value = str(value)
+        r_dict[value] = r_dict[value] + (index + 1,)
+    for elem in r_dict:
+        r_dict[elem] = sum(r_dict[elem]) / len(r_dict[elem])
+
+    for value in seq1:
+        value = str(value)
+        r1[value] = r1[value] + r_dict[value]
+    
+    for value in seq2:
+        value = str(value)
+        r2[value] = r2[value] + r_dict[value]
+
+    u1 = (len_seq1 * len_seq2) + len_seq1 * (len_seq1 + 1) / 2 - sum(r1.values())
+    u2 = (len_seq1 * len_seq2) + len_seq2 * (len_seq2 + 1) / 2 - sum(r2.values())
+
+    u_stat = min(u1, u2)
+
+    if u_stat <= 127:
+        return False
+    
+    return True
+
+
+def decomposition(seq: str) -> list:
+    """
+    Decomposes 3 letter-sequence into list of single aminoacids.
+    Used in `check_and_procees_seq` function if sequence uses 3-letter abbreviation.
+    """
+    len_seq, dec_seq = len(seq), []
+    if len(seq) % 3 == 0:
+        for i in range(0, len_seq, 3):
+            dec_seq.append(seq[i:i+3].lower().capitalize())
+    
+    return dec_seq
+
+
+def seq_transform(seq: list) -> str:
+    """
+    Transforms aminoacid abbreviation format from 3-letter into 1-letter.
+    Used in `check_and_procees_seq` function if sequence uses 3-letter abbreviation.
+    """
+    seq_tr = ''
+    for aa in seq:
+        seq_tr += AA_TR_DICT[aa]
+    
+    return seq_tr
+  
+  
+
+def aa_content_check(seq: str) -> dict:
+    "Returns aminoacids content of the protein"
+    seq_content = dict.fromkeys(AA_UNIPROT_CONTENT.keys(), 0)
+    for AAcd in seq.upper():

```

Comment:
В Python переменные (за исключением констант) пишут прописными буквами.
```suggestion
    for aa_cd in seq.upper():
```